packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -261,6 +269,9 @@ export class Stream extends EventEmitter {
       msSinceLastPacket: msSinceLastPacket,
       amount: data.amount.toString()
     })
+    this._assetCode = data.assetCode
```

Comment:
It might be better to use total sent and total delivered rather than an individual packet. I did note that there were some outlier packets where the exchangeRate can be considerably different than the average. 

I assume this may be later used for tipping UI showing destination currencies ? 

I see currently that tips are using a separate GraphQL mutation and are not really using the streaming method. 

Why not send the actual amounts sent from the tipping endpoint ? Is the endpoint async ?

Reply:
@sharafian made me aware of the fact that the webmonetization standard defines the events with the destination currency and hence the tip event should also state values in the destination currency. We have been discussing whether we should change the standard now that STREAM receipts already contain the destination amounts. It may be desirable to show the user how much they sent instead of what the website received.

> Why not send the actual amounts sent from the tipping endpoint ? Is the endpoint async ?

I'm not sure I understand what you mean.

Reply:
> Why not send the actual amounts sent from the tipping endpoint ? Is the endpoint async ?
So it's a QraphQL mutation endpoint that does the heavy lifting of sending money. Does that endpoint return asynchronously or does it wait until the full send is complete. If the latter you should know exactly how much was sent (source) and received right ?